packages/withdrawals-contract/schema/withdrawals-documents.json

```diff
@@ -0,0 +1,92 @@
+{
+  "withdrawal": {
+    "description": "Withdrawal document to track underlying transaction",
+    "type": "object",
+    "indices": [
+      {
+        "name": "pooling",
+        "properties": [
+          {
+            "status": "asc"
+          },
+          {
+            "pooling": "asc"
+          },
+          {
+            "coreFeePerByte": "asc"
+          },
+          {
+            "$updatedAt": "asc"
+          }
+        ],
+        "unique": false
+      },
+      {
+        "name": "transaction",
+        "properties": [
+          {
+            "transactionId": "asc"
+          },
+          {
+            "status": "asc"
+          }
+        ],
+        "unique": true
+      }
+    ],
+    "properties": {
+      "transactionId": {
+        "description": "Hash of asset unlock transaction. Populated with untied transaction hash on “pooled” stage and with signed transaction on “broatcasted” stage",
+        "type": "array",
+        "byteArray": true,
+        "minItems": 32,
+        "maxItems": 32
+      },
+      "amount": {
+        "type": "integer",
+        "description": "The amount to be withdrawn",
+        "minimum": 1000
+      },
+      "coreFeePerByte": {
+        "type": "integer",
+        "description": "This is the fee that you are willing to spend for this transaction in Duffs/Byte",
+        "minimum": 1
+      },
+      "pooling": {
+        "type": "integer",
+        "description": "This indicated the level at which Platform should try to pool this transaction",
+        "enum": [
+          0
+        ]
+      },
+      "outputScript": {
+        "type": "array",
+        "byteArray": true,
+        "minItems": 23,
+        "maxItems": 25
+      },
+      "status": {
+        "type": "integer",
+        "enum": [
+          0,
+          1,
+          2,
+          3,
+          4
+        ]
+      }
+    },
+    "minProperties": 6,
+    "additionalProperties": false,
+    "required": [
+      "$createdAt",
+      "$updatedAt",
+      "transactionId",
```

Comment:
It's optional, as far I remember

Reply:
You can't make `transactionId` optional, as it is a part of compound index

Reply:
For non-unique indices it should be possible. Please investigate.